lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -349,12 +349,32 @@ static bool isBinaryOp(linalg::GenericOp linalgOp) {
   return false;
 }
 
-// Return true if the operands are all the same.
-static bool allOperandsHaveSameType(linalg::GenericOp linalgOp) {
+// Returns true if all the operands of the current linalgOp are shape type and
+// have the same statically-known shape. For example: A: memref<56x32xf32,
+// strided<[32, 1], offset: ?>> B: memref<56x32xf32> allOperandsHaveSameShape(A,
+// B) return true. C: memref<1x32xf32> allOperandsHaveSameShape(A, C) return
+// false.
+static bool allOperandsHaveSameShape(linalg::GenericOp linalgOp) {
```

Comment:
Is it still valid when strides are dynamic?

Reply:
Or different/dynamic offsets?

Reply:
The offset in the example is dynamic.